packages/fuels-code-gen/src/program_bindings/abigen/bindings/function_generator.rs

```diff
@@ -108,7 +117,13 @@ impl From<&FunctionGenerator> for TokenStream {
         let arg_declarations = fun.args.iter().map(|component| {
             let name = &component.field_name;
             let field_type = &component.field_type;
-            quote! { #name: #field_type }
+
+            find_into_impl_type(&field_type.type_name.to_string())
+                .map(|new_type| {
+                    let new_type = new_type;
```

Comment:
Seems odd, the compiler gave you trouble?

Reply:
good catch!